core/shared/src/main/scala/org/http4s/Message.scala

```diff
@@ -621,6 +621,25 @@ object Request {
     val UnixSocketAddress: Key[UnixSocketAddress] =
       Key.newKey[SyncIO, UnixSocketAddress].unsafeRunSync()
   }
+
+  implicit class RequestOps[F[_]](request: Request[F]) {
+
+    /** cURL representation of the request and a new request
+      *
+      * Supported cURL-Parameters are: --request, --url, --header --data.
+      * Note that `asCurlWith` will print the request body, which may have privacy implications.
+      *
+      * This method will consume the body create a new request with the cached body to avoid issues
+      * this may cause, which may consume large amounts of memory to cache a request that streams a
+      * large amount of data.
+      */
+    def asCurlWithBody(redactHeadersWhen: CIString => Boolean = Headers.SensitiveHeaders.contains)(
```

Comment:
Just to confirm I understand, this is the same as what was added to 0.23 in https://github.com/http4s/http4s/commit/62a19ca68fd0444d2426a94a17cb2c710f576a54
but is wrapped in an implicit class to help handle that `Concurrent` is invariant, but `Request` is now covariant?

Reply:
Ross better knows, but as far as I understand, yeah it's because `Request` is covariant.

Reply:
Correct.